euclidlib/photo/_le3_2pcf_wl.py

```diff
@@ -51,15 +52,35 @@ def correlation_functions(path: str | PathLike[str]) -> dict[_DictKey, NDArray[A
       for the output dictionary.
     """
 
-    xi: dict[_DictKey, NDArray[Any]] = {}
+    xi: dict[_DictKey, Result] = {}
     with fitsio.FITS(path) as fits:
-        for hdu in fits:
-            if "2D" not in hdu.get_extname():
-                continue
-            key = _key_from_string(hdu.get_extname())
+        for hdu in fits[1:]:
+            extname = hdu.get_extname()
+            key = _key_from_string(extname)
             if key is None:
                 continue
-            xi[key] = hdu.read()
+            data = hdu.read()
+            THETA = data["THETA"]
+            bin_size = np.log(THETA[1]) - np.log(THETA[0])
+            half_bins = np.exp(0.5 * bin_size)
+            LOWER = THETA / half_bins
+            UPPER = THETA * half_bins
```

Comment:
Since the data is not naturally binned, we shouldn't introduce a binning ourselves in the reading routine